src/main/java/org/opensearch/flowframework/workflow/RegisterLocalModelStep.java

```diff
@@ -0,0 +1,196 @@
+import org.opensearch.ml.common.model.MLModelFormat;
+import org.opensearch.ml.common.model.TextEmbeddingModelConfig;
+import org.opensearch.ml.common.model.TextEmbeddingModelConfig.FrameworkType;
+import org.opensearch.ml.common.model.TextEmbeddingModelConfig.TextEmbeddingModelConfigBuilder;
+import org.opensearch.ml.common.transport.register.MLRegisterModelInput;
+import org.opensearch.ml.common.transport.register.MLRegisterModelInput.MLRegisterModelInputBuilder;
+import org.opensearch.ml.common.transport.register.MLRegisterModelResponse;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+import static org.opensearch.flowframework.common.CommonValue.ALL_CONFIG;
+import static org.opensearch.flowframework.common.CommonValue.DESCRIPTION_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.EMBEDDING_DIMENSION;
+import static org.opensearch.flowframework.common.CommonValue.FRAMEWORK_TYPE;
+import static org.opensearch.flowframework.common.CommonValue.MODEL_CONTENT_HASH_VALUE;
+import static org.opensearch.flowframework.common.CommonValue.MODEL_FORMAT;
+import static org.opensearch.flowframework.common.CommonValue.MODEL_GROUP_ID;
+import static org.opensearch.flowframework.common.CommonValue.MODEL_ID;
+import static org.opensearch.flowframework.common.CommonValue.MODEL_TYPE;
+import static org.opensearch.flowframework.common.CommonValue.NAME_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.REGISTER_MODEL_STATUS;
+import static org.opensearch.flowframework.common.CommonValue.URL;
+import static org.opensearch.flowframework.common.CommonValue.VERSION_FIELD;
+
+/**
+ * Step to register a local model
+ */
+public class RegisterLocalModelStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(RegisterLocalModelStep.class);
+
+    private MachineLearningNodeClient mlClient;
+
+    static final String NAME = "register_local_model";
+
+    /**
+     * Instantiate this class
+     * @param mlClient client to instantiate MLClient
+     */
+    public RegisterLocalModelStep(MachineLearningNodeClient mlClient) {
+        this.mlClient = mlClient;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
+
+        CompletableFuture<WorkflowData> registerLocalModelFuture = new CompletableFuture<>();
+
+        ActionListener<MLRegisterModelResponse> actionListener = new ActionListener<>() {
+            @Override
+            public void onResponse(MLRegisterModelResponse mlRegisterModelResponse) {
+                logger.info("Local Model registration successful");
+                registerLocalModelFuture.complete(
+                    new WorkflowData(
+                        Map.ofEntries(
+                            Map.entry(MODEL_ID, mlRegisterModelResponse.getModelId()),
```

Comment:
Can you verify once the register API present on `feature/agent_dev` branch is returning model_id and not task_id. If it returns taskId then we might need a GetTaskStep to get the model id from task id.

Reply:
Good call out, the `MLRegisterModelResponse` has an optional `modelD` field, but the `TransportMLRegisterModelAction` just responds with the taskId [here](https://github.com/opensearch-project/ml-commons/blob/7e44eb282f0d3f6deeb07993d83a00bcfcd5edd5/plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java#L267C35-L267C35) after it dispatches the task to another node. I can add the ml task step in my next commit

Reply:
A month old work I did before.
```java
/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 */
package org.opensearch.flowframework.workflow.Model;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.opensearch.client.Client;
import org.opensearch.client.node.NodeClient;
import org.opensearch.flowframework.client.MLClient;
import org.opensearch.flowframework.workflow.WorkflowData;
import org.opensearch.flowframework.workflow.WorkflowStep;
import org.opensearch.ml.client.MachineLearningNodeClient;
import org.opensearch.ml.common.MLTask;

import java.io.IOException;
import java.util.List;
import java.util.Map;
import java.util.Map.Entry;
import java.util.concurrent.CompletableFuture;

public class GetTaskStep implements WorkflowStep {

    private static final Logger logger = LogManager.getLogger(GetTaskStep.class);

    private Client client;

    private final String NAME = "get_task";
    private static final String TASK_ID = "task_id";

    public GetTaskStep(Client client) {
        this.client = client;
    }

    @Override
    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
        CompletableFuture<WorkflowData> getTaskFuture = new CompletableFuture<>();

        String taskId = null;

        for (WorkflowData workflowData : data) {
            Map<String, Object> content = workflowData.getContent();

            for (Entry<String, Object> entry : content.entrySet()) {
                if (entry.getKey() == TASK_ID) {
                    taskId = (String) content.get(TASK_ID);
                    break;
                }
            }

        }

        if (taskId != null) {
            MachineLearningNodeClient machineLearningNodeClient = MLClient.createMLClient((NodeClient) client);

            MLTask mlTaskResponse = machineLearningNodeClient.getTask(taskId).actionGet();

            getTaskFuture.complete(new WorkflowData(Map.of("modelId", mlTaskResponse.getModelId())));
        } else {
            logger.error("Failed to fetch model");
            getTaskFuture.completeExceptionally(new IOException("Failed to fetch model"));
        }

        return getTaskFuture;
    }

    @Override
    public String getName() {
        return NAME;
    }
}
```